### PR TITLE
Fix plugin related issues

### PR DESCRIPTION
**Pull request by ashu8912**
This PR fixes and adds a few things that are required for the plugins to work correctly.

1. The ```createRouteURL``` method in ```frontend/src/lib/router.tsx``` doesn't considers routes from redux store which could be a problem when a plugin wants to render a link and the only way the plugin can communicate it's route params is through the store so we have to make the createRouteURL method aware of the store routes.
2. We need the Editor component exported because it will come in useful when we want to modify a kube object.
3. We need  APIs such as ```useLocation``` or ```useHistory``` etc. from react-router package so it makes sense to export it as well

### CODE REVIEW

File: frontend/src/lib/router.tsx [hunk 1]
```diff
@@ -431,8 +431,12 @@ export interface RouteURLProps {
...
 }
 
-export function createRouteURL(routeName: string, params: RouteURLProps = {}) {
-  const route = getRoute(routeName);
+export function createRouteURL(
+  routeName: string,
+  params: RouteURLProps = {},
+  storeRoutes?: { [path: string]: any }
+) {
+  const route = getRoute(routeName) || (storeRoutes && storeRoutes[routeName]);
```
[the remark this answers is not in the document]
Comment (ashu8912) [reply frontend/src/lib/router.tsx]: Fixed

**Review by joaquimrocha — APPROVED** [review]

**Comment by joaquimrocha** [comment]
@illume , do you want to give one last look at this one?

File: frontend/src/lib/router.tsx [hunk 2]
```diff
@@ -432,7 +433,8 @@ export interface RouteURLProps {
 }
 
 export function createRouteURL(routeName: string, params: RouteURLProps = {}) {
-  const route = getRoute(routeName);
+  const storeRoutes = store.getState().ui.routes;
+  const route = getRoute(routeName) || (storeRoutes && storeRoutes[routeName]);
```
Comment (illume) [inline frontend/src/lib/router.tsx:437]: I wonder why getRoute(routeName) is chosen over the store routes here?

**Comment by illume** [comment]
I'm not sure what I should be testing. The PR could use a description and the PR title could mention which issues it addresses.

File: frontend/src/lib/router.tsx (hunk 2 above)
Comment (ashu8912) [reply frontend/src/lib/router.tsx:437]: So it's like first search in the list of routes we have if that's undefined check if it's in storeRoutes

**Comment by ashu8912** [comment]
@illume i have added a description for the PR

**Pull request merged**